app/repository/AdminDataUnitLinksEditRepository.scala

```diff
@@ -1,15 +1,12 @@
 package repository
 
-import uk.gov.ons.sbr.models.{ Period, VatRef }
+import repository.rest.PatchStatus
+import uk.gov.ons.sbr.models.{ Period, UnitId, VatRef }
 import uk.gov.ons.sbr.models.edit.Patch
 
 import scala.concurrent.Future
 
-sealed trait EditParentLinkStatus
-case object EditSuccess extends EditParentLinkStatus
-case object EditFailure extends EditParentLinkStatus
-case object EditConflict extends EditParentLinkStatus
-
 trait AdminDataUnitLinksEditRepository {
```

Comment:
It still feels to me as if the Patch is being created too soon.
If we decided to change the mechanism by which sbr-control-api accepts edits, we would either have to change the interface of the UnitLinksEditRepository, or provide an implementation that is able to parse a Patch to extract what it needs - even though it is not now using Patch.

Ideally, the interface for this repository should accept a raw description of what to do:
```def createChildUnitLink(unitKey: UnitKey: childType: UnitType, childId: UnitId)```

We can then freely substitute different implementations of a repository without impacting any code above this.  Our current implementation would create a patch based on these arguments (using your ```PatchCreation``` utility and forwarding to sbr-control-api).

In this way, the only thing that knows that sbr-control-api is using patch is our internal implementation of the UnitLinkRepository.